test/e2e/ingress.go

```diff
@@ -135,6 +143,32 @@ var _ = framework.KubeDescribe("Loadbalancing: L7 [Feature:Ingress]", func() {
 			By("should reject HTTP traffic")
 			ExpectNoError(pollURL(fmt.Sprintf("http://%v/", ip), "", lbPollTimeout, httpClient, true))
 
+			By("should have correct firewall rule for ingress")
+			fw, err := getFwForIngress(cloudConfig)
+			Expect(err).NotTo(HaveOccurred())
+			// Retrieve nodeTags from the node instance as the target tags for the ingress firewall rule
+			nodes := framework.GetReadySchedulableNodesOrDie(jig.client)
+			Expect(len(nodes.Items) > 0).Should(BeTrue())
+			nodeTags, err := getInstanceTags(cloudConfig, nodes.Items[0].Name)
```

Comment:
as a general rule for this file:     

* the jig encapsulates any Kubernets/talking-to-apiserver logic, so looking up the service, parsing out ports etc. 
* the specific controller (be it gce or nginx) encapsulates gce or nginx specific logic (eg: listing firewalls etc)

So I suggest partitioning all the naked functions you have into one of those two buckets as appropriate